ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -606,3 +675,80 @@ decodeShelleyResult query = case query of
     GetStakePoolParams {}                      -> fromCBOR
     GetRewardInfoPools                         -> fromCBOR
     GetPoolState {}                            -> fromCBOR
+    GetStakeSnapshots {}                       -> fromCBOR
+
+-- | The stake snapshot returns information about the mark, set, go ledger snapshots for a pool,
+-- plus the total active stake for each snapshot that can be used in a 'sigma' calculation.
+--
+-- Each snapshot is taken at the end of a different era. The go snapshot is the current one and
+-- was taken two epochs earlier, set was taken one epoch ago, and mark was taken immediately
+-- before the start of the current epoch.
+data StakeSnapshot crypto = StakeSnapshot
+  { ssMarkPool :: !SL.Coin
+  , ssSetPool  :: !SL.Coin
+  , ssGoPool   :: !SL.Coin
+  } deriving (Eq, Show, Generic)
+
+instance NFData (StakeSnapshot crypto)
+
+instance
+  Crypto crypto =>
+  ToCBOR (StakeSnapshot crypto)
+  where
+  toCBOR
+    StakeSnapshot
+    { ssMarkPool
+    , ssSetPool
+    , ssGoPool
+    } = encodeListLen 3
+      <> toCBOR ssMarkPool
+      <> toCBOR ssSetPool
+      <> toCBOR ssGoPool
+
+instance
+  Crypto crypto =>
+  FromCBOR (StakeSnapshot crypto)
+  where
+  fromCBOR = do
+    enforceSize "StakeSnapshot" 3
+    StakeSnapshot
+      <$> fromCBOR
+      <*> fromCBOR
+      <*> fromCBOR
+
+data StakeSnapshots crypto = StakeSnapshots
+  { ssStakeSnapshots  :: Map (SL.KeyHash 'SL.StakePool crypto) (StakeSnapshot crypto)
```

Comment:
This is the strict `Map`, so I think this field should also be strict.

Reply:
Fixed!